Build failure: postgis

### Steps To Reproduce

Steps to reproduce the behavior:

1.  nix-build -A postgresql_16.pkgs.postgis

### Build log

```
  "___gxx_personality_v0", referenced from:
      _flatgeobuf_encode_header in libflatgeobuf.la(flatgeobuf_c.o)
      _flatgeobuf_encode_feature in libflatgeobuf.la(flatgeobuf_c.o)
      _flatgeobuf_create_index in libflatgeobuf.la(flatgeobuf_c.o)
      flatbuffers::FlatBufferBuilder::~FlatBufferBuilder() in libflatgeobuf.la(flatgeobuf_c.o)
      std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(flatgeobuf_c.o)
      _lwgeom_wagyu_clip_by_box in libwagyu.la(lwgeom_wagyu.o)
      bool mapbox::geometry::wagyu::wagyu<int>::execute<int>(mapbox::geometry::wagyu::clip_type, mapbox::geometry::multi_polygon<int, std::__1::vector>&, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
      ...
ld: symbol(s) not found for architecture arm64
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [/nix/store/vdca86kb8jvcpv3cc9lg63v3071gcsf4-postgresql-16.1/lib/pgxs/src/makefiles/../../src/Makefile.shlib:278: postgis-3.dylib] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-postgis-3.4.0.drv-0/postgis-3.4.0/postgis'
make: *** [GNUmakefile:25: all] Error 1
error: builder for '/nix/store/faq0j20rx0hhallzi2nx410z4m8zd1ab-postgis-3.4.0.drv' failed with exit code 2;
       last 10 log lines:
       >       flatbuffers::FlatBufferBuilder::~FlatBufferBuilder() in libflatgeobuf.la(flatgeobuf_c.o)
       >       std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(flatgeobuf_c.o)
       >       _lwgeom_wagyu_clip_by_box in libwagyu.la(lwgeom_wagyu.o)
       >       bool mapbox::geometry::wagyu::wagyu<int>::execute<int>(mapbox::geometry::wagyu::clip_type, mapbox::geometry::multi_polygon<int, std::__1::vector>&, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
       >       ...
       > ld: symbol(s) not found for architecture arm64
       > clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
       > make[1]: *** [/nix/store/vdca86kb8jvcpv3cc9lg63v3071gcsf4-postgresql-16.1/lib/pgxs/src/makefiles/../../src/Makefile.shlib:278: postgis-3.dylib] Error 1
       > make[1]: Leaving directory '/private/tmp/nix-build-postgis-3.4.0.drv-0/postgis-3.4.0/postgis'
       > make: *** [GNUmakefile:25: all] Error 1
```
Full build log at https://gist.github.com/purcell/5032e82a71ad29c816632726ee058af5

### Additional context


### Notify maintainers

@NixOS/geospatial @MarcWeber 

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.1.0, macOS 14.1.1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(steve): `"nixpkgs"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/Users/steve/.nix-defexpr/channels/nixpkgs

### Priorities

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

I can reproduce this, but I'm way in over my head, I don't think I can help fix it. I'll put down everything I found out, hopefully it will help someone else.

It seems all the undined symbols are from the C++ standard library. The common solution for an error like that is to add `-lc++` to the linker flags, but checking the linker invocation shows that it is already present. (twice, in fact)

See below for the full invocation of the compiler and linker for the failing file:

<details>
```
feuh@W6WFDWT3VX:/tmp/postgis/postgis-3.4.0/postgis$ cc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Werror=unguarded-availability-new -Wendif-labels -Wmissing-format-attribute -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-unused-command-line-argument -Wno-compound-token-split-by-macro -Wno-cast-function-type-strict -g -O2  -fvisibility=hidden -bundle -o postgis-3.dylib postgis_module.o lwgeom_accum.o lwgeom_union.o lwgeom_spheroid.o lwgeom_ogc.o lwgeom_functions_analytic.o lwgeom_inout.o lwgeom_functions_basic.o lwgeom_btree.o lwgeom_box.o lwgeom_box3d.o lwgeom_geos.o lwgeom_geos_prepared.o lwgeom_geos_clean.o lwgeom_geos_relatematch.o lwgeom_generate_grid.o lwgeom_export.o lwgeom_in_gml.o lwgeom_in_kml.o lwgeom_in_marc21.o lwgeom_out_marc21.o lwgeom_in_geohash.o lwgeom_in_geojson.o lwgeom_in_encoded_polyline.o lwgeom_triggers.o lwgeom_dump.o lwgeom_dumppoints.o lwgeom_functions_lrs.o lwgeom_functions_temporal.o lwgeom_rectree.o long_xact.o lwgeom_sqlmm.o lwgeom_rtree.o lwgeom_transform.o lwgeom_window.o gserialized_typmod.o gserialized_gist_2d.o gserialized_gist_nd.o gserialized_supportfn.o gserialized_spgist_2d.o gserialized_spgist_3d.o gserialized_spgist_nd.o brin_2d.o brin_nd.o brin_common.o gserialized_estimate.o geography_inout.o geography_btree.o geography_centroid.o geography_measurement.o geography_measurement_trees.o geometry_inout.o postgis_libprotobuf.o vector_tile.pb-c.o geobuf.pb-c.o mvt.o lwgeom_out_mvt.o geobuf.o lwgeom_out_geobuf.o lwgeom_out_geojson.o flatgeobuf.o lwgeom_in_flatgeobuf.o lwgeom_out_flatgeobuf.o postgis_legacy.o -lm -fvisibility=hidden ../deps/wagyu/libwagyu.la -lc++ ../deps/flatgeobuf/libflatgeobuf.la -lc++ ../libpgcommon/libpgcommon.a ../liblwgeom/.libs/liblwgeom.a  -L/nix/store/r3ci57b9h44dm8bzgddj2s33gmzf0k9d-geos-3.11.2/lib -lgeos_c -L/nix/store/crykfdf8mmkll3a524l34cd80cxczqrm-proj-9.3.0/lib -lproj -L/nix/store/wvy3a51f7nhy0k441j0n1cz8hnpynvkr-json-c-0.16/lib -ljson-c -L/nix/store/phw2l0iwb43pd3j1hlaml46asziw7vw0-protobuf-c-unstable-2023-07-08/lib -lprotobuf-c -L/nix/store/al2fcmdcsxlx9zv5nsr4464dmrypf2bk-libxml2-2.11.5/lib -lxml2 -L/nix/store/a377d66g1rgsd4vkzxdhkza5g55fkmky-zlib-1.3/lib -lz -lc++  -lm -bundle_loader /nix/store/vdca86kb8jvcpv3cc9lg63v3071gcsf4-postgresql-16.1/bin/postgres -v
clang version 16.0.6
Target: arm64-apple-darwin
Thread model: posix
InstalledDir: /nix/store/a5v30qll5i02vr9y97bk1rdx3mm6kvlm-clang-16.0.6/bin
 "/nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/bin/ld" -demangle -lto_library /nix/store/a5v30qll5i02vr9y97bk1rdx3mm6kvlm-clang-16.0.6/lib/libLTO.dylib -dynamic -arch arm64 -bundle -bundle_loader /nix/store/vdca86kb8jvcpv3cc9lg63v3071gcsf4-postgresql-16.1/bin/postgres -platform_version macos 11.0.0 11.0.0 -o postgis-3.dylib -L/nix/store/r3ci57b9h44dm8bzgddj2s33gmzf0k9d-geos-3.11.2/lib -L/nix/store/crykfdf8mmkll3a524l34cd80cxczqrm-proj-9.3.0/lib -L/nix/store/wvy3a51f7nhy0k441j0n1cz8hnpynvkr-json-c-0.16/lib -L/nix/store/phw2l0iwb43pd3j1hlaml46asziw7vw0-protobuf-c-unstable-2023-07-08/lib -L/nix/store/al2fcmdcsxlx9zv5nsr4464dmrypf2bk-libxml2-2.11.5/lib -L/nix/store/a377d66g1rgsd4vkzxdhkza5g55fkmky-zlib-1.3/lib -L/nix/store/wvy3a51f7nhy0k441j0n1cz8hnpynvkr-json-c-0.16/lib -L/nix/store/jxgh20m58s1nwzficc9h8iz9rbaszzbl-libxcrypt-4.4.36/lib -L/nix/store/vf7k7l7ny35g1nmninaqjbryl2xqkag5-libcxx-16.0.6/lib -L/nix/store/x622mzi1g7v7nh7zq5r8d0nxbhrg9px3-libcxxabi-16.0.6/lib -L/nix/store/l92ls1vba62zsywdj81hlgmgx5adn6bn-compiler-rt-libc-16.0.6/lib -L/nix/store/a377d66g1rgsd4vkzxdhkza5g55fkmky-zlib-1.3/lib -L/nix/store/g5r20rs0qhcjcbf9dhbnbd9ksg0h0jmx-libiconv-50/lib -L/nix/store/al2fcmdcsxlx9zv5nsr4464dmrypf2bk-libxml2-2.11.5/lib -L/nix/store/vdca86kb8jvcpv3cc9lg63v3071gcsf4-postgresql-16.1/lib -L/nix/store/8gk2lrcblimgy9d1zabpz1sr0fvsqvcz-postgresql-16.1-lib/lib -L/nix/store/r3ci57b9h44dm8bzgddj2s33gmzf0k9d-geos-3.11.2/lib -L/nix/store/crykfdf8mmkll3a524l34cd80cxczqrm-proj-9.3.0/lib -L/nix/store/d43agl398v9dss19ciacycz4mk2bj5h6-gdal-3.7.3/lib -L/nix/store/wvy3a51f7nhy0k441j0n1cz8hnpynvkr-json-c-0.16/lib -L/nix/store/phw2l0iwb43pd3j1hlaml46asziw7vw0-protobuf-c-unstable-2023-07-08/lib -L/nix/store/13g4cgq3mhdl364f27r10xasyr6108ri-pcre2-10.42/lib -L/nix/store/bj25mj5bpvbx03gqzf7hbpk6n7w4khr6-libobjc-11.0.0/lib -L/nix/store/jxgh20m58s1nwzficc9h8iz9rbaszzbl-libxcrypt-4.4.36/lib -L/nix/store/vf7k7l7ny35g1nmninaqjbryl2xqkag5-libcxx-16.0.6/lib -L/nix/store/x622mzi1g7v7nh7zq5r8d0nxbhrg9px3-libcxxabi-16.0.6/lib -L/nix/store/l92ls1vba62zsywdj81hlgmgx5adn6bn-compiler-rt-libc-16.0.6/lib -L/nix/store/a377d66g1rgsd4vkzxdhkza5g55fkmky-zlib-1.3/lib -L/nix/store/g5r20rs0qhcjcbf9dhbnbd9ksg0h0jmx-libiconv-50/lib -L/nix/store/al2fcmdcsxlx9zv5nsr4464dmrypf2bk-libxml2-2.11.5/lib -L/nix/store/vdca86kb8jvcpv3cc9lg63v3071gcsf4-postgresql-16.1/lib -L/nix/store/8gk2lrcblimgy9d1zabpz1sr0fvsqvcz-postgresql-16.1-lib/lib -L/nix/store/r3ci57b9h44dm8bzgddj2s33gmzf0k9d-geos-3.11.2/lib -L/nix/store/crykfdf8mmkll3a524l34cd80cxczqrm-proj-9.3.0/lib -L/nix/store/d43agl398v9dss19ciacycz4mk2bj5h6-gdal-3.7.3/lib -L/nix/store/wvy3a51f7nhy0k441j0n1cz8hnpynvkr-json-c-0.16/lib -L/nix/store/phw2l0iwb43pd3j1hlaml46asziw7vw0-protobuf-c-unstable-2023-07-08/lib -L/nix/store/13g4cgq3mhdl364f27r10xasyr6108ri-pcre2-10.42/lib -L/nix/store/bj25mj5bpvbx03gqzf7hbpk6n7w4khr6-libobjc-11.0.0/lib -L/nix/store/g1w6cr7w95mpny3pg0j4jwwlmwqbmdzm-libSystem-11.0.0/lib -L/nix/store/yl2yc5pjp29aqdphfyav7vka3ysdm9x4-clang-16.0.6-lib/lib -L/nix/store/vf7k7l7ny35g1nmninaqjbryl2xqkag5-libcxx-16.0.6/lib -no_uuid postgis_module.o lwgeom_accum.o lwgeom_union.o lwgeom_spheroid.o lwgeom_ogc.o lwgeom_functions_analytic.o lwgeom_inout.o lwgeom_functions_basic.o lwgeom_btree.o lwgeom_box.o lwgeom_box3d.o lwgeom_geos.o lwgeom_geos_prepared.o lwgeom_geos_clean.o lwgeom_geos_relatematch.o lwgeom_generate_grid.o lwgeom_export.o lwgeom_in_gml.o lwgeom_in_kml.o lwgeom_in_marc21.o lwgeom_out_marc21.o lwgeom_in_geohash.o lwgeom_in_geojson.o lwgeom_in_encoded_polyline.o lwgeom_triggers.o lwgeom_dump.o lwgeom_dumppoints.o lwgeom_functions_lrs.o lwgeom_functions_temporal.o lwgeom_rectree.o long_xact.o lwgeom_sqlmm.o lwgeom_rtree.o lwgeom_transform.o lwgeom_window.o gserialized_typmod.o gserialized_gist_2d.o gserialized_gist_nd.o gserialized_supportfn.o gserialized_spgist_2d.o gserialized_spgist_3d.o gserialized_spgist_nd.o brin_2d.o brin_nd.o brin_common.o gserialized_estimate.o geography_inout.o geography_btree.o geography_centroid.o geography_measurement.o geography_measurement_trees.o geometry_inout.o postgis_libprotobuf.o vector_tile.pb-c.o geobuf.pb-c.o mvt.o lwgeom_out_mvt.o geobuf.o lwgeom_out_geobuf.o lwgeom_out_geojson.o flatgeobuf.o lwgeom_in_flatgeobuf.o lwgeom_out_flatgeobuf.o postgis_legacy.o -lm ../deps/wagyu/libwagyu.la -lc++ ../deps/flatgeobuf/libflatgeobuf.la -lc++ ../libpgcommon/libpgcommon.a ../liblwgeom/.libs/liblwgeom.a -lgeos_c -lproj -ljson-c -lprotobuf-c -lxml2 -lz -lc++ -lm -liconv -arch arm64 -lSystem /nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/resource-root/lib/darwin/libclang_rt.osx.a -F/nix/store/7aiqcq3qjyy3cx3c2qafcv1q559hgbbc-apple-framework-CoreFoundation-11.0.0/Library/Frameworks -F/nix/store/7aiqcq3qjyy3cx3c2qafcv1q559hgbbc-apple-framework-CoreFoundation-11.0.0/Library/Frameworks
Undefined symbols for architecture arm64:
  "std::exception::what() const", referenced from:
      vtable for std::__1::bad_function_call in libflatgeobuf.la(packedrtree.o)
  "std::length_error::~length_error()", referenced from:
      std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(flatgeobuf_c.o)
      std::__1::__throw_length_error[abi:v160006](char const*) in libwagyu.la(lwgeom_wagyu.o)
      std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(packedrtree.o)
      std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(geometrywriter.o)
  "std::runtime_error::~runtime_error()", referenced from:
      bool mapbox::geometry::wagyu::wagyu<int>::execute<int>(mapbox::geometry::wagyu::clip_type, mapbox::geometry::multi_polygon<int, std::__1::vector>&, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::add_ring_to_local_minima_list<int>(std::__1::vector<mapbox::geometry::wagyu::edge<int>, std::__1::allocator<mapbox::geometry::wagyu::edge<int>>>&, std::__1::deque<mapbox::geometry::wagyu::local_minimum<int>, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>>>&, mapbox::geometry::wagyu::polygon_type) in libwagyu.la(lwgeom_wagyu.o)
      mapbox::geometry::wagyu::hp_intersection_swap<int>::operator()(mapbox::geometry::wagyu::bound<int>* const&, mapbox::geometry::wagyu::bound<int>* const&) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::process_edges_at_top_of_scanbeam<int>(int, std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>&, std::__1::vector<int, std::__1::allocator<int>>&, std::__1::vector<mapbox::geometry::wagyu::local_minimum<int>*, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>*>> const&, std::__1::vector<mapbox::geometry::wagyu::local_minimum<int>*, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>*>>::iterator&, mapbox::geometry::wagyu::ring_manager<int>&, mapbox::geometry::wagyu::clip_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::process_intersect_list<int>(std::__1::vector<mapbox::geometry::wagyu::intersect_node<int>, std::__1::allocator<mapbox::geometry::wagyu::intersect_node<int>>>&, mapbox::geometry::wagyu::clip_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::ring_manager<int>&, std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>&) in libwagyu.la(lwgeom_wagyu.o)
      mapbox::geometry::wagyu::on_intersection_swap<int>::operator()(mapbox::geometry::wagyu::bound<int>* const&, mapbox::geometry::wagyu::bound<int>* const&) in libwagyu.la(lwgeom_wagyu.o)
      std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>::iterator mapbox::geometry::wagyu::do_maxima<int>(std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>::iterator&, std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>::iterator&, mapbox::geometry::wagyu::clip_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::ring_manager<int>&, std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>&) in libwagyu.la(lwgeom_wagyu.o)
      ...
  "std::overflow_error::~overflow_error()", referenced from:
      FlatGeobuf::PackedRTree::generateLevelBounds(unsigned long long, unsigned short) in libflatgeobuf.la(packedrtree.o)
      FlatGeobuf::PackedRTree::size(unsigned long long, unsigned short) in libflatgeobuf.la(packedrtree.o)
  "std::invalid_argument::~invalid_argument()", referenced from:
      FlatGeobuf::PackedRTree::init(unsigned short) in libflatgeobuf.la(packedrtree.o)
      FlatGeobuf::PackedRTree::generateLevelBounds(unsigned long long, unsigned short) in libflatgeobuf.la(packedrtree.o)
      FlatGeobuf::PackedRTree::size(unsigned long long, unsigned short) in libflatgeobuf.la(packedrtree.o)
  "std::bad_array_new_length::bad_array_new_length()", referenced from:
      std::__throw_bad_array_new_length[abi:v160006]() in libflatgeobuf.la(flatgeobuf_c.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libwagyu.la(lwgeom_wagyu.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libflatgeobuf.la(packedrtree.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libflatgeobuf.la(geometrywriter.o)
  "std::bad_array_new_length::~bad_array_new_length()", referenced from:
      std::__throw_bad_array_new_length[abi:v160006]() in libflatgeobuf.la(flatgeobuf_c.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libwagyu.la(lwgeom_wagyu.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libflatgeobuf.la(packedrtree.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libflatgeobuf.la(geometrywriter.o)
  "std::exception::~exception()", referenced from:
      std::__1::bad_function_call::~bad_function_call() in libflatgeobuf.la(packedrtree.o)
      std::__1::bad_function_call::~bad_function_call() in libflatgeobuf.la(packedrtree.o)
  "std::terminate()", referenced from:
      ___clang_call_terminate in libflatgeobuf.la(flatgeobuf_c.o)
  "typeinfo for std::length_error", referenced from:
      std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(flatgeobuf_c.o)
      std::__1::__throw_length_error[abi:v160006](char const*) in libwagyu.la(lwgeom_wagyu.o)
      std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(packedrtree.o)
      std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(geometrywriter.o)
  "typeinfo for std::runtime_error", referenced from:
      bool mapbox::geometry::wagyu::wagyu<int>::execute<int>(mapbox::geometry::wagyu::clip_type, mapbox::geometry::multi_polygon<int, std::__1::vector>&, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::add_ring_to_local_minima_list<int>(std::__1::vector<mapbox::geometry::wagyu::edge<int>, std::__1::allocator<mapbox::geometry::wagyu::edge<int>>>&, std::__1::deque<mapbox::geometry::wagyu::local_minimum<int>, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>>>&, mapbox::geometry::wagyu::polygon_type) in libwagyu.la(lwgeom_wagyu.o)
      mapbox::geometry::wagyu::hp_intersection_swap<int>::operator()(mapbox::geometry::wagyu::bound<int>* const&, mapbox::geometry::wagyu::bound<int>* const&) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::process_edges_at_top_of_scanbeam<int>(int, std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>&, std::__1::vector<int, std::__1::allocator<int>>&, std::__1::vector<mapbox::geometry::wagyu::local_minimum<int>*, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>*>> const&, std::__1::vector<mapbox::geometry::wagyu::local_minimum<int>*, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>*>>::iterator&, mapbox::geometry::wagyu::ring_manager<int>&, mapbox::geometry::wagyu::clip_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::process_intersect_list<int>(std::__1::vector<mapbox::geometry::wagyu::intersect_node<int>, std::__1::allocator<mapbox::geometry::wagyu::intersect_node<int>>>&, mapbox::geometry::wagyu::clip_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::ring_manager<int>&, std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>&) in libwagyu.la(lwgeom_wagyu.o)
      mapbox::geometry::wagyu::on_intersection_swap<int>::operator()(mapbox::geometry::wagyu::bound<int>* const&, mapbox::geometry::wagyu::bound<int>* const&) in libwagyu.la(lwgeom_wagyu.o)
      std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>::iterator mapbox::geometry::wagyu::do_maxima<int>(std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>::iterator&, std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>::iterator&, mapbox::geometry::wagyu::clip_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::ring_manager<int>&, std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>&) in libwagyu.la(lwgeom_wagyu.o)
      ...
  "typeinfo for std::overflow_error", referenced from:
      FlatGeobuf::PackedRTree::generateLevelBounds(unsigned long long, unsigned short) in libflatgeobuf.la(packedrtree.o)
      FlatGeobuf::PackedRTree::size(unsigned long long, unsigned short) in libflatgeobuf.la(packedrtree.o)
  "typeinfo for std::invalid_argument", referenced from:
      FlatGeobuf::PackedRTree::init(unsigned short) in libflatgeobuf.la(packedrtree.o)
      FlatGeobuf::PackedRTree::generateLevelBounds(unsigned long long, unsigned short) in libflatgeobuf.la(packedrtree.o)
      FlatGeobuf::PackedRTree::size(unsigned long long, unsigned short) in libflatgeobuf.la(packedrtree.o)
  "typeinfo for std::bad_array_new_length", referenced from:
      std::__throw_bad_array_new_length[abi:v160006]() in libflatgeobuf.la(flatgeobuf_c.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libwagyu.la(lwgeom_wagyu.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libflatgeobuf.la(packedrtree.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libflatgeobuf.la(geometrywriter.o)
  "typeinfo for std::exception", referenced from:
      typeinfo for std::__1::bad_function_call in libflatgeobuf.la(packedrtree.o)
  "vtable for __cxxabiv1::__class_type_info", referenced from:
      typeinfo for std::__1::__function::__base<void (unsigned char*, unsigned long)> in libflatgeobuf.la(flatgeobuf_c.o)
      typeinfo for flatgeobuf_create_index::$_0 in libflatgeobuf.la(flatgeobuf_c.o)
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for __cxxabiv1::__si_class_type_info", referenced from:
      typeinfo for std::__1::__shared_ptr_emplace<FeatureItem, std::__1::allocator<FeatureItem>> in libflatgeobuf.la(flatgeobuf_c.o)
      typeinfo for std::__1::__function::__func<flatgeobuf_create_index::$_0, std::__1::allocator<flatgeobuf_create_index::$_0>, void (unsigned char*, unsigned long)> in libflatgeobuf.la(flatgeobuf_c.o)
      typeinfo for std::__1::bad_function_call in libflatgeobuf.la(packedrtree.o)
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for std::length_error", referenced from:
      std::length_error::length_error[abi:v160006](char const*) in libflatgeobuf.la(flatgeobuf_c.o)
      std::length_error::length_error[abi:v160006](char const*) in libwagyu.la(lwgeom_wagyu.o)
      std::length_error::length_error[abi:v160006](char const*) in libflatgeobuf.la(packedrtree.o)
      std::length_error::length_error[abi:v160006](char const*) in libflatgeobuf.la(geometrywriter.o)
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for std::overflow_error", referenced from:
      std::overflow_error::overflow_error[abi:v160006](char const*) in libflatgeobuf.la(packedrtree.o)
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for std::invalid_argument", referenced from:
      std::invalid_argument::invalid_argument[abi:v160006](char const*) in libflatgeobuf.la(packedrtree.o)
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "operator delete[](void*)", referenced from:
      _flatgeobuf_encode_header in libflatgeobuf.la(flatgeobuf_c.o)
      _flatgeobuf_encode_feature in libflatgeobuf.la(flatgeobuf_c.o)
      _flatgeobuf_create_index in libflatgeobuf.la(flatgeobuf_c.o)
      FlatGeobuf::PackedRTree::~PackedRTree() in libflatgeobuf.la(flatgeobuf_c.o)
      flatbuffers::vector_downward::reallocate(unsigned long) in libflatgeobuf.la(flatgeobuf_c.o)
      flatbuffers::FlatBufferBuilder::~FlatBufferBuilder() in libflatgeobuf.la(flatgeobuf_c.o)
      flatbuffers::vector_downward::reallocate(unsigned long) in libflatgeobuf.la(geometrywriter.o)
      ...
  "operator delete(void*)", referenced from:
      _flatgeobuf_encode_header in libflatgeobuf.la(flatgeobuf_c.o)
      _flatgeobuf_encode_feature in libflatgeobuf.la(flatgeobuf_c.o)
      FlatGeobuf::GeometryWriter::~GeometryWriter() in libflatgeobuf.la(flatgeobuf_c.o)
      _flatgeobuf_create_index in libflatgeobuf.la(flatgeobuf_c.o)
      FlatGeobuf::PackedRTree::~PackedRTree() in libflatgeobuf.la(flatgeobuf_c.o)
      std::__1::vector<std::__1::shared_ptr<FlatGeobuf::Item>, std::__1::allocator<std::__1::shared_ptr<FlatGeobuf::Item>>>::~vector[abi:v160006]() in libflatgeobuf.la(flatgeobuf_c.o)
      flatbuffers::FlatBufferBuilder::~FlatBufferBuilder() in libflatgeobuf.la(flatgeobuf_c.o)
      ...
  "operator new[](unsigned long)", referenced from:
      flatbuffers::vector_downward::reallocate(unsigned long) in libflatgeobuf.la(flatgeobuf_c.o)
      FlatGeobuf::PackedRTree::init(unsigned short) in libflatgeobuf.la(packedrtree.o)
      flatbuffers::vector_downward::reallocate(unsigned long) in libflatgeobuf.la(geometrywriter.o)
  "operator new(unsigned long)", referenced from:
      _flatgeobuf_encode_header in libflatgeobuf.la(flatgeobuf_c.o)
      _flatgeobuf_create_index in libflatgeobuf.la(flatgeobuf_c.o)
      void std::__1::vector<std::__1::shared_ptr<FlatGeobuf::Item>, std::__1::allocator<std::__1::shared_ptr<FlatGeobuf::Item>>>::__push_back_slow_path<std::__1::shared_ptr<FlatGeobuf::Item>>(std::__1::shared_ptr<FlatGeobuf::Item>&&) in libflatgeobuf.la(flatgeobuf_c.o)
      std::__1::__function::__func<flatgeobuf_create_index::$_0, std::__1::allocator<flatgeobuf_create_index::$_0>, void (unsigned char*, unsigned long)>::__clone() const in libflatgeobuf.la(flatgeobuf_c.o)
      lwpoly_to_vwgpoly(LWPOLY const*, GBOX const*) in libwagyu.la(lwgeom_wagyu.o)
      ptarray_to_wglinearring(POINTARRAY const*) in libwagyu.la(lwgeom_wagyu.o)
      void std::__1::vector<mapbox::geometry::linear_ring<int, std::__1::vector>, std::__1::allocator<mapbox::geometry::linear_ring<int, std::__1::vector>>>::__push_back_slow_path<mapbox::geometry::linear_ring<int, std::__1::vector>>(mapbox::geometry::linear_ring<int, std::__1::vector>&&) in libwagyu.la(lwgeom_wagyu.o)
      ...
  "operator new(unsigned long, std::nothrow_t const&)", referenced from:
      void mapbox::geometry::wagyu::build_hot_pixels<int>(std::__1::deque<mapbox::geometry::wagyu::local_minimum<int>, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>>>&, mapbox::geometry::wagyu::ring_manager<int>&) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::execute_vatti<int>(std::__1::deque<mapbox::geometry::wagyu::local_minimum<int>, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>>>&, mapbox::geometry::wagyu::ring_manager<int>&, mapbox::geometry::wagyu::clip_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::correct_topology<int>(mapbox::geometry::wagyu::ring_manager<int>&) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::process_intersections<int>(int, std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>&, mapbox::geometry::wagyu::clip_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::ring_manager<int>&) in libwagyu.la(lwgeom_wagyu.o)
      std::__1::vector<mapbox::geometry::wagyu::ring<int>*, std::__1::allocator<mapbox::geometry::wagyu::ring<int>*>> mapbox::geometry::wagyu::sort_rings_smallest_to_largest<int>(mapbox::geometry::wagyu::ring_manager<int>&) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::assign_new_ring_parents<int>(mapbox::geometry::wagyu::ring_manager<int>&, mapbox::geometry::wagyu::ring<int>*, std::__1::vector<mapbox::geometry::wagyu::ring<int>*, std::__1::allocator<mapbox::geometry::wagyu::ring<int>*>>&) in libwagyu.la(lwgeom_wagyu.o)
      std::__1::vector<mapbox::geometry::wagyu::point<int>*, std::__1::allocator<mapbox::geometry::wagyu::point<int>*>> mapbox::geometry::wagyu::sort_ring_points<int>(mapbox::geometry::wagyu::ring<int>*) in libwagyu.la(lwgeom_wagyu.o)
      ...
  "___cxa_allocate_exception", referenced from:
      std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(flatgeobuf_c.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libflatgeobuf.la(flatgeobuf_c.o)
      bool mapbox::geometry::wagyu::wagyu<int>::execute<int>(mapbox::geometry::wagyu::clip_type, mapbox::geometry::multi_polygon<int, std::__1::vector>&, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
      std::__1::__throw_length_error[abi:v160006](char const*) in libwagyu.la(lwgeom_wagyu.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::add_ring_to_local_minima_list<int>(std::__1::vector<mapbox::geometry::wagyu::edge<int>, std::__1::allocator<mapbox::geometry::wagyu::edge<int>>>&, std::__1::deque<mapbox::geometry::wagyu::local_minimum<int>, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>>>&, mapbox::geometry::wagyu::polygon_type) in libwagyu.la(lwgeom_wagyu.o)
      mapbox::geometry::wagyu::hp_intersection_swap<int>::operator()(mapbox::geometry::wagyu::bound<int>* const&, mapbox::geometry::wagyu::bound<int>* const&) in libwagyu.la(lwgeom_wagyu.o)
      ...
  "___cxa_begin_catch", referenced from:
      ___clang_call_terminate in libflatgeobuf.la(flatgeobuf_c.o)
      _lwgeom_wagyu_clip_by_box in libwagyu.la(lwgeom_wagyu.o)
  "___cxa_end_catch", referenced from:
      _lwgeom_wagyu_clip_by_box in libwagyu.la(lwgeom_wagyu.o)
  "___cxa_free_exception", referenced from:
      std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(flatgeobuf_c.o)
      bool mapbox::geometry::wagyu::wagyu<int>::execute<int>(mapbox::geometry::wagyu::clip_type, mapbox::geometry::multi_polygon<int, std::__1::vector>&, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
      std::__1::__throw_length_error[abi:v160006](char const*) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::add_ring_to_local_minima_list<int>(std::__1::vector<mapbox::geometry::wagyu::edge<int>, std::__1::allocator<mapbox::geometry::wagyu::edge<int>>>&, std::__1::deque<mapbox::geometry::wagyu::local_minimum<int>, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>>>&, mapbox::geometry::wagyu::polygon_type) in libwagyu.la(lwgeom_wagyu.o)
      mapbox::geometry::wagyu::hp_intersection_swap<int>::operator()(mapbox::geometry::wagyu::bound<int>* const&, mapbox::geometry::wagyu::bound<int>* const&) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::process_edges_at_top_of_scanbeam<int>(int, std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>&, std::__1::vector<int, std::__1::allocator<int>>&, std::__1::vector<mapbox::geometry::wagyu::local_minimum<int>*, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>*>> const&, std::__1::vector<mapbox::geometry::wagyu::local_minimum<int>*, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>*>>::iterator&, mapbox::geometry::wagyu::ring_manager<int>&, mapbox::geometry::wagyu::clip_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::process_intersect_list<int>(std::__1::vector<mapbox::geometry::wagyu::intersect_node<int>, std::__1::allocator<mapbox::geometry::wagyu::intersect_node<int>>>&, mapbox::geometry::wagyu::clip_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::ring_manager<int>&, std::__1::vector<mapbox::geometry::wagyu::bound<int>*, std::__1::allocator<mapbox::geometry::wagyu::bound<int>*>>&) in libwagyu.la(lwgeom_wagyu.o)
      ...
  "___cxa_throw", referenced from:
      std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(flatgeobuf_c.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libflatgeobuf.la(flatgeobuf_c.o)
      bool mapbox::geometry::wagyu::wagyu<int>::execute<int>(mapbox::geometry::wagyu::clip_type, mapbox::geometry::multi_polygon<int, std::__1::vector>&, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
      std::__1::__throw_length_error[abi:v160006](char const*) in libwagyu.la(lwgeom_wagyu.o)
      std::__throw_bad_array_new_length[abi:v160006]() in libwagyu.la(lwgeom_wagyu.o)
      void mapbox::geometry::wagyu::add_ring_to_local_minima_list<int>(std::__1::vector<mapbox::geometry::wagyu::edge<int>, std::__1::allocator<mapbox::geometry::wagyu::edge<int>>>&, std::__1::deque<mapbox::geometry::wagyu::local_minimum<int>, std::__1::allocator<mapbox::geometry::wagyu::local_minimum<int>>>&, mapbox::geometry::wagyu::polygon_type) in libwagyu.la(lwgeom_wagyu.o)
      mapbox::geometry::wagyu::hp_intersection_swap<int>::operator()(mapbox::geometry::wagyu::bound<int>* const&, mapbox::geometry::wagyu::bound<int>* const&) in libwagyu.la(lwgeom_wagyu.o)
      ...
  "___gxx_personality_v0", referenced from:
      _flatgeobuf_encode_header in libflatgeobuf.la(flatgeobuf_c.o)
      _flatgeobuf_encode_feature in libflatgeobuf.la(flatgeobuf_c.o)
      _flatgeobuf_create_index in libflatgeobuf.la(flatgeobuf_c.o)
      flatbuffers::FlatBufferBuilder::~FlatBufferBuilder() in libflatgeobuf.la(flatgeobuf_c.o)
      std::__1::__throw_length_error[abi:v160006](char const*) in libflatgeobuf.la(flatgeobuf_c.o)
      _lwgeom_wagyu_clip_by_box in libwagyu.la(lwgeom_wagyu.o)
      bool mapbox::geometry::wagyu::wagyu<int>::execute<int>(mapbox::geometry::wagyu::clip_type, mapbox::geometry::multi_polygon<int, std::__1::vector>&, mapbox::geometry::wagyu::fill_type, mapbox::geometry::wagyu::fill_type) in libwagyu.la(lwgeom_wagyu.o)
      ...
ld: symbol(s) not found for architecture arm64
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
```
</details>

I tried to go back and find the latest version that still builds. FIrst idea was e6f61d425ba2fc702f8d57a017a68996b202a4fb, one commit before the upgrade to 3.4.0 was merged. There was no postgres_16 then, but with postgres_14 it builds fine:

```
$ git checkout e6f61d425ba2fc702f8d57a017a68996b202a4fb
$ nix-build -A postgresql_14.pkgs.postgis
/nix/store/5d1wljssashfavzg53k8q81d3jrm8l3q-postgis-3.3.3
```

Next suspect was 23e69f92c8a5d3898d4cb51373479382f3a9b10e, which was the upgrade to 3.4.0. That one was cached, so the upgrade itself is not the culprit:

```
$ nix-build -A postgresql_14.pkgs.postgis
this path will be fetched (1.77 MiB download, 41.36 MiB unpacked):
  /nix/store/wdkibb03rn956jirmadwsn27l48rmf1r-postgis-3.4.0
copying path '/nix/store/wdkibb03rn956jirmadwsn27l48rmf1r-postgis-3.4.0' from 'https://cache.nixos.org'...
/nix/store/wdkibb03rn956jirmadwsn27l48rmf1r-postgis-3.4.0
```

Next one was 46423a1a750594236673c1d741def4e93cf5a8f7, the commit where `postgres_16` was first added. It built fine as well, so postgres_16 isn't the culprit either.

Then I tried current master, but with postgres_14. That also fails with the exact same error as above.

I tried to start a bisect, but that requires rebuilding *everything* multiple times as nothing is cached.

Just checking the hydra builds, I can see that [build 241085829](https://hydra.nixos.org/build/241085829) as first to fail and [build 240806714]https://hydra.nixos.org/build/240806714) was last to succeed, but between those two are over 1400 commits. This was just a few days ago.

Bisecting is also difficult because netcdf, one of postgis' dependencies, also fails to build sometimes, so you have to do it manually to differentiate between those two kinds of failures.

--- a/pkgs/servers/sql/postgresql/ext/postgis.nix
+++ b/pkgs/servers/sql/postgresql/ext/postgis.nix
@@ -31,7 +31,10 @@ stdenv.mkDerivation rec {
   dontDisableStatic = true;
 
   # postgis config directory assumes /include /lib from the same root for json-c library
-  NIX_LDFLAGS = "-L${lib.getLib json_c}/lib";
+  NIX_LDFLAGS = "-L${lib.getLib json_c}/lib"
+    # Work around https://github.com/NixOS/nixpkgs/issues/166205.
+    + lib.optionalString (stdenv.cc.isClang && stdenv.cc.libcxx != null) " -l${stdenv.cc.libcxx.cxxabi.libName}";
+
 
   preConfigure = ''
     sed -i 's@/usr/bin/file@${file}/bin/file@' configure

